core/src/main/java/com/google/instrumentation/trace/PropagationUtil.java

```diff
@@ -0,0 +1,230 @@
+
+  /**
+   * Serializes a {@link SpanContext} using the HTTP standard format.
+   *
+   * @param spanContext the {@code SpanContext} to serialize.
+   * @return the serialized US-ASCII encoded HTTP header value.
+   * @throws NullPointerException if the {@code spanContext} is null.
+   */
+  public static String toHttpHeaderValue(SpanContext spanContext) {
+    return BaseEncoding.base16().encode(toBinaryValue(spanContext));
+  }
+
+  /**
+   * Parses the {@link SpanContext} from the HTTP standard format.
+   *
+   * @param input a US-ASCII encoded buffer of characters from which the {@code SpanContext} will be
+   *     parsed.
+   * @return the parsed {@code SpanContext}.
+   * @throws NullPointerException if the {@code input} is null.
+   * @throws IllegalArgumentException if the {@code input} is invalid.
+   * @throws IllegalStateException if the version is not supported.
+   */
+  public static SpanContext fromHttpHeaderValue(CharSequence input) {
+    return fromBinaryValue(BaseEncoding.base16().decode(input));
+  }
+
+  /**
+   * Serializes a {@link SpanContext} using the binary format.
+   *
+   * @param spanContext the {@code SpanContext} to serialize.
+   * @return the serialized binary value.
+   * @throws NullPointerException if the {@code spanContext} is null.
+   */
+  public static byte[] toBinaryValue(SpanContext spanContext) {
+    checkNotNull(spanContext, "spanContext");
+    return handler.toBinaryFormat(spanContext);
+  }
+
+  /**
+   * Parses the {@link SpanContext} from the binary format.
+   *
+   * @param bytes a binary encoded buffer from which the {@code SpanContext} will be parsed.
+   * @return the parsed {@code SpanContext}.
+   * @throws NullPointerException if the {@code input} is null.
+   * @throws IllegalArgumentException if the {@code input} is invalid.
+   * @throws IllegalStateException if the version is not supported.
+   */
+  public static SpanContext fromBinaryValue(byte[] bytes) {
+    checkNotNull(bytes, "bytes");
+    return handler.fromBinaryFormat(bytes);
+  }
+
+  /**
+   * Sets a new {@link Handler}.
+   *
+   * @param newHandler the new {@code Handler} to be set.
+   */
+  public static void setHandler(Handler newHandler) {
+    handler = newHandler;
+  }
+
+  /** Abstract class that allows implementation of the new version format support. */
+  public abstract static class Handler {
+    /**
+     * Serializes a {@link SpanContext} using the binary format.
+     *
+     * @param spanContext the {@code SpanContext} to serialize.
+     * @return the serialized binary value.
+     */
+    public abstract byte[] toBinaryFormat(SpanContext spanContext);
+
+    /**
+     * Parses the {@link SpanContext} from the binary format.
+     *
+     * @param bytes a binary encoded buffer from which the {@code SpanContext} will be parsed.
+     * @return the parsed {@code SpanContext}.
+     */
+    public abstract SpanContext fromBinaryFormat(byte[] bytes);
```

Comment:
Would it be better to use `InputStream` and `OutputStream` for these two methods, so that they are more general?  That is what `StatsSerializer` does: https://github.com/google/instrumentation-java/blob/2e16e06b33d259591117383eb59552ffa15950d2/core_impl/src/main/java/com/google/instrumentation/stats/StatsSerializer.java#L50-89